source/ns_cmdline.c

```diff
@@ -344,6 +350,18 @@ void cmd_free(void)
     cmd.mutex_wait_fnc = NULL;
     cmd.mutex_release_fnc = NULL;
 }
+
+bool cmd_redirect(char *redir_buf, const size_t redir_len)
+{
+    if (cmd.redirecting || redir_buf == NULL || redir_len == 0) {
```

Comment:
Looks like `cmd.redirecting` check may be `true` only due to coding error. Are assert guards in use in this project?